subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -1522,3 +1522,32 @@ int nrf_cloud_parse_cell_pos_response(const char *const buf,
 	cJSON_Delete(cell_pos_obj);
 	return ret;
 }
+
+bool nrf_cloud_detect_disconnection_request(const char *const buf)
+{
+	if (buf == NULL) {
+		return false;
+	}
+
+	/* Fast test to avoid parsing EVERY message with cJSON. */
+	if (strstr(buf, NRF_CLOUD_JSON_APPID_VAL_DEVICE) == NULL ||
```

Comment:
Maybe add a note here that justifies the use of non-length input based functions such as `strstr` and `cJSON_Parse`. Due to the explicit null termination in https://github.com/nrfconnect/sdk-nrf/blob/53a4486ed22b28f0fcbf16dbcfbeca0cf672e3f2/subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c#L855

Reply:
Actually, it may be simpler (and better) to not assume null termination since arbitrary binary may potentially be passed to this function, good point.

Yes, it's true that the MQTT transport guarantees all output is null-terminated, but it is better not to rely on that.

Good catch, thank you.

Reply:
I have updated the code to verify that there is a null byte within the first 200 characters of the passed buffer. This should be sufficient, since the JSON messages we are looking out for are typically about 150 characters. Anything not terminated within 200 characters is either not a null-terminated string, or is otherwise not a DISCON message